server/genquery2/src/genquery2_sql.cpp

```diff
@@ -0,0 +1,1218 @@
+        }
+
+        throw std::invalid_argument{fmt::format("table [{}] not supported", _table_name)};
+    } // to_index
+    // clang-format on
+
+    // clang-format off
+    constexpr auto table_edges = std::to_array<edge_type>({
+        {to_index("R_COLL_MAIN"), to_index("R_DATA_MAIN")},                 // R_COLL_MAIN.coll_id = R_DATA_MAIN.coll_id
+        {to_index("R_COLL_MAIN"), to_index("R_OBJT_ACCESS")},               // R_COLL_MAIN.coll_id = R_OBJT_ACCESS.object_id
+        {to_index("R_COLL_MAIN"), to_index("R_OBJT_METAMAP")},              // R_COLL_MAIN.coll_id = R_OBJT_METAMAP.object_id
+        {to_index("R_COLL_MAIN"), to_index("R_TICKET_MAIN")},               // R_COLL_MAIN.coll_id = R_TICKET_MAIN.object_id
+
+        {to_index("R_DATA_MAIN"), to_index("R_OBJT_ACCESS")},               // R_DATA_MAIN.data_id = R_OBJT_ACCESS.object_id
+        {to_index("R_DATA_MAIN"), to_index("R_OBJT_METAMAP")},              // R_DATA_MAIN.data_id = R_OBJT_METAMAP.object_id
+        {to_index("R_DATA_MAIN"), to_index("R_RESC_MAIN")},                 // R_DATA_MAIN.resc_id = R_RESC_MAIN.resc_id
+        {to_index("R_DATA_MAIN"), to_index("R_TICKET_MAIN")},               // R_DATA_MAIN.data_id = R_TICKET_MAIN.object_id
+
+        {to_index("R_META_MAIN"), to_index("R_OBJT_METAMAP")},              // R_META_MAIN.meta_id = R_OBJT_METAMAP.meta_id
+
+        {to_index("R_OBJT_ACCESS"), to_index("R_TOKN_MAIN")},               // R_OBJT_ACCESS.access_type_id = R_TOKN_MAIN.token_id
+
+        {to_index("R_OBJT_METAMAP"), to_index("R_RESC_MAIN")},              // R_OBJT_METAMAP.object_id = R_RESC_MAIN.resc_id
+        {to_index("R_OBJT_METAMAP"), to_index("R_USER_MAIN")},              // R_OBJT_METAMAP.object_id = R_USER_MAIN.user_id
+
+        {to_index("R_TICKET_MAIN"), to_index("R_USER_MAIN")},               // R_TICKET_MAIN.user_id = R_USER_MAIN.user_id
+
+        {to_index("R_TICKET_MAIN"), to_index("R_TICKET_ALLOWED_HOSTS")},    // R_TICKET_MAIN.ticket_id = R_TICKET_ALLOWED_HOSTS.ticket_id
+        {to_index("R_TICKET_MAIN"), to_index("R_TICKET_ALLOWED_USERS")},    // R_TICKET_MAIN.ticket_id = R_TICKET_ALLOWED_USERS.ticket_id
+        {to_index("R_TICKET_MAIN"), to_index("R_TICKET_ALLOWED_GROUPS")},   // R_TICKET_MAIN.ticket_id = R_TICKET_ALLOWED_GROUPS.ticket_id
+
+        {to_index("R_USER_MAIN"), to_index("R_USER_AUTH")},                 // R_USER_MAIN.user_id = R_USER_AUTH.user_id
+        {to_index("R_USER_MAIN"), to_index("R_USER_GROUP")},                // R_USER_MAIN.user_id = R_USER_GROUP.group_user_id
+        {to_index("R_USER_MAIN"), to_index("R_USER_PASSWORD")},             // R_USER_MAIN.user_id = R_USER_PASSWORD.user_id
+        {to_index("R_USER_MAIN"), to_index("R_USER_SESSION_KEY")},          // R_USER_MAIN.user_id = R_USER_SESSION_KEY.user_id
+
+        // TODO Handle R_USER_GROUP?
+        // TODO Handle R_QUOTA_MAIN
+        // TODO Handle R_QUOTA_USAGE
+    }); // table_edges
+    // clang-format on
+
+    // clang-format off
```

Comment:
Is this needed?

Reply:
I'm guessing you're asking due to L140?

I could remove them, but I wanted to guard against future devs inserting code between the blocks.

Reply:
I'll double check whether these are necessary.

Reply:
Leaving these in for the reason I mentioned earlier.

Attempting to remove the clang-format directives altogether screws up the formatting, so that's a no-go.